x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/http/PublishableHttpResource.java

```diff
@@ -317,7 +320,10 @@ protected void putResource(final RestClient client,
 
 
         final Request request = new Request("PUT", resourceBasePath + "/" + resourceName);
-        addParameters(request);
+        addDefaultParameters(request);
+        if (parameters != null) {
```

Comment:
I tend to prefer using empty collections instead of making parameters nullable (we could just use `Collections.emptyMap()` in all the calls that don't need additional parameters).